arkouda/dataframe.py

```diff
@@ -1845,6 +1848,261 @@ def numeric_help(d):
         ret_dict = json.loads(generic_msg(cmd="corrMatrix", args=args))
         return DataFrame({c: create_pdarray(ret_dict[c]) for c in self.columns})
 
+    @typechecked
+    def register(self, user_defined_name: str) -> DataFrame:
+        """
+        Register this DataFrame object and underlying components with the Arkouda server
+
+        Parameters
```

Comment:
This is more a quirk having to do with dataframe having some `inplace` functionality than anything to do with register functionality, but we should prob let the user know that any `inplace` changes made to a dataframe once it's been registered will not appear in an attached copy. I'm not sure how to note this, @Ethan-DeBandi99 what do u think